### DISCUSSION TITLE
I Prefer FTP/SFTP Uploads Done In A Sorted Order; Makes Tracking Much Easier

### DISCUSSION THREAD

**Discussion by WhatImListnTo** [iterate-ch/cyberduck#16109] [Ideas]
I am currently performing a multiple GiB upload of directories, sub-directories, and files from my local computer to an FTP server.  It appears that the list of files that gets uploaded is unsorted and happens most likely due to physical layout on the source disk.  This makes it difficult to track progress.  Here is a screenshot of my server tree command of the destination directory:

![07-08-2024_SERVER_TREE](https://github.com/iterate-ch/cyberduck/assets/55523642/c86b96a0-b71f-4daf-a745-78c70192a9e7)

The parent directory is Hubrid.  Directory "2019, Dark War" stopped upload at file "27-Aggression (Feat. Hubrid)".  This is incomplete.  There are more files, and it also skipped over nine directories below it.  I would prefer if the upload list were sorted similar to how tree is sorting the output.  Process files in a sorted order to make tracking and checking progress on the server side easier.

**Comment by WhatImListnTo** [comment]
People may prefer how the upload currently operates, so I suggest making this an option.

**Comment by WhatImListnTo** [comment]
My SFTP upload is now working on artist names in the J's.  Cyberduck completed Hubrid/"2019, Dark War" for all files.  However, the nine empty directories remain empty.  I hope they eventually get uploaded.  If the job finishes and they remain empty, the debug logs would have been lost.  Is this log space configurable?  I have tons of free space, and there is no need to keep only five log ZIPs.

**Reply by AliveDevil** [reply]
You can configure `logging.archives` by creating a file `default.properties` in `%AppData%\Cyberduck`.
i.e. the default is:
```
logging.archives=5
```

**Comment by WhatImListnTo** [comment]
It is uploading the nine missing right now.  A use case for my sorted upload is allowing for upload recovery for failed attempts.  This is helpful when unexpected data gets uploaded.  For example, I started this upload last week Friday, and it continues today after several issues.  I was able to verify what was uploaded to the server, move files to correct directories to match my source, and start a new SFTP upload that was a continuation of what I had originally wanted to upload.  This allowed me to save time by skipping what was already uploaded.  Uploading in order makes this possible.
